src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/BinderInvocation.cs

```diff
@@ -1,84 +1,72 @@
+                IsCandidateInvocationOperation(operation))
             {
-                return null;
+                return new BinderInvocation()
+                {
+                    CandidateOperation = operation,
+                    Location = invocationSyntax.GetLocation()
+                };
             }
 
-            return new BinderInvocation()
-            {
-                Operation = operation,
-                Location = invocationSyntax.GetLocation()
-            };
+            return default;
         }
 
-        private static bool IsCandidateInvocationExpressionSyntax(SyntaxNode node, out InvocationExpressionSyntax? invocationSyntax)
+        public static bool IsCandidateSyntaxNode(SyntaxNode node)
         {
-            if (node is InvocationExpressionSyntax
-                {
-                    Expression: MemberAccessExpressionSyntax
-                    {
-                        Name.Identifier.ValueText: string memberName
-                    }
-                } syntax && IsCandidateBindingMethodName(memberName))
+            return node is InvocationExpressionSyntax
             {
-                invocationSyntax = syntax;
-                return true;
-            }
-
-            invocationSyntax = null;
-            return false;
+                Expression: MemberAccessExpressionSyntax
+                {
+                    Name.Identifier.ValueText: string memberName
+                }
+            } && IsCandidateBindingMethodName(memberName);
 
             static bool IsCandidateBindingMethodName(string name) =>
                 IsCandidateMethodName_ConfigurationBinder(name) ||
                 IsCandidateMethodName_OptionsBuilderConfigurationExtensions(name) ||
                 IsValidMethodName_OptionsConfigurationServiceCollectionExtensions(name);
         }
 
-        private static bool IsCandidateInvocation(IInvocationOperation operation)
+        private static bool IsCandidateInvocationOperation(IInvocationOperation operation)
         {
             if (operation.TargetMethod is not IMethodSymbol
                 {
                     IsExtensionMethod: true,
                     Name: string methodName,
-                    ContainingType: ITypeSymbol
-                    {
-                        Name: string containingTypeName,
-                        ContainingNamespace: INamespaceSymbol { } containingNamespace,
-                    } containingType
-                } method ||
-                containingNamespace.ToDisplayString() is not string containingNamespaceName)
+                    ContainingType.Name: string containingTypeName,
+                })
             {
                 return false;
             }
 
             return (containingTypeName) switch
             {
                 "ConfigurationBinder" =>
-                    containingNamespaceName is "Microsoft.Extensions.Configuration" &&
```

Comment:
Don't we still need to check the namespaces of the symbols?

Reply:
Thought we might save on the `ToDisplayString()` alloc; I think the chances of a namespace match are high if the other checks are `true`.

Reply:
It's ok to skip those checks if you were doing an early check on the syntax node, but this is the method that checks against the symbol. You still need to check you're in the right namespace somewhere.

Reply:
Reverting to allocating the display string.